torch/lib/c10d/test/TCPStoreTest.cpp

```diff
@@ -136,10 +142,89 @@ void testHelper(const std::string& prefix = "") {
   }
 }
 
+void testWatchKeyCallback(const std::string& prefix = "") {
+  // Callback function increments counter of the total number of callbacks that
+  // were run
+  int numCallbacksExecuted = 0;
+  std::function<void(std::string, std::string)> callback =
+      [&numCallbacksExecuted](
+          const std::string& old_value, const std::string& new_value) {
```

Comment:
We should also validate we are receiving the right `old_value` and `new_value` in the callbacks as well.